src/core/Akka.Remote.Tests/Transport/AkkaProtocolSpec.cs

```diff
@@ -405,6 +407,52 @@ public void ProtocolStateActor_must_handle_correctly_when_the_handler_is_registe
             });
         }
 
+        [Fact]
+        public void ProtocolStateActor_must_give_up_outbound_after_connection_timeout()
```

Comment:
Ported specs to ensure that handshake timeouts are implemented correctly on both inbound and outbound side.